presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -106,7 +107,9 @@ public void checkCanShowSchemas(ConnectorSecurityContext context)
     @Override
     public Set<String> filterSchemas(ConnectorSecurityContext context, Set<String> schemaNames)
     {
-        return schemaNames;
+        return schemaNames.stream()
+                .filter(schemaName -> checkSchemaAnyPermission(context, schemaName))
```

Comment:
it does not look correct. Schema is visible when user is an owner of schema or has any access to any table within this schema.

Reply:
so, further check tables rule ?

Reply:
> further check tables rule ?

I think so.

Reply:
@kokosing ready for review.